src/printer.js

```diff
@@ -908,8 +907,10 @@ function genericPrintNoParens(path, options, print, args) {
       return nodeStr(n, options);
     case "UnaryExpression":
       parts.push(n.operator);
-      if ( n.argument && n.argument.type !== "UnaryExpression" && ( n.operator !== '+' && n.operator !== '-') ) {
-        parts.push(' ');
+
+      if (/[a-z]$/.test(n.operator)) parts.push(" ");
```

Comment:
what is this catching other than typeof?
Shouldn't it also be an `else if` instead of an `if` with the conditional below it.